### PR TITLE
[ceos]: add support to use ceos as neighbors

### PR DESCRIPTION
**Pull request by lguohan**
### Description of PR

Summary:
    ceos is container-based eos. ceos has smaller cpu and memory
    footprint than veos, around 1G memory consumption per container.

Fixes # (issue)

### Type of change

- [] Bug fix
- [x] Testbed and Framework(new/improvement)
- [] Test case(new/improvement)

#### How did you verify/test it?
```
lgh@jenkins-worker-15:~/sonic-mgmt/ansible$ docker ps
CONTAINER ID        IMAGE                                                        COMMAND                  CREATED             STATUS              PORTS               NAMES
8d0bdeb5c1b8        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   46 minutes ago      Up About an hour                        ceos_vms6-1_VM0123
09d2b0351784        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   47 minutes ago      Up About an hour                        ceos_vms6-1_VM0122
f05c342172c1        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   47 minutes ago      Up About an hour                        ceos_vms6-1_VM0121
b46a5a5fb628        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   47 minutes ago      Up About an hour                        ceos_vms6-1_VM0120
c32fb412748a        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   47 minutes ago      Up About an hour                        ceos_vms6-1_VM0119
323ebc3fdb4b        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   48 minutes ago      Up About an hour                        ceos_vms6-1_VM0118
c358776a894a        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   48 minutes ago      Up About an hour                        ceos_vms6-1_VM0117
66a79165b1fb        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   48 minutes ago      Up About an hour                        ceos_vms6-1_VM0116
d61584224f28        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   49 minutes ago      Up About an hour                        ceos_vms6-1_VM0115
123ea09d51d5        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   49 minutes ago      Up About an hour                        ceos_vms6-1_VM0114
f88f773fa64d        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   49 minutes ago      Up About an hour                        ceos_vms6-1_VM0113
cbdb6929c81d        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0112
bbf6d63d638e        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0111
7cd72105f204        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0110
8fa22f152f7e        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0109
96009d48e0e2        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0108
b06fb75c2b18        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0107
df4c5ec496dd        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0106
21f43c4e6b3f        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0105
366d76c11d3c        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0104
b6d9eeb33950        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0103
c8949ce351fd        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0102
d80a43050d3e        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   About an hour ago   Up About an hour                        ceos_vms6-1_VM0101
f2ddee44594f        sonicdev-microsoft.azurecr.io:443/docker-ptf:latest          "/usr/local/bin/supe…"   2 hours ago         Up 2 hours                              ptf_vms6-1
85019935bb34        ceosimage:4.23.2F                                            "/sbin/init systemd.…"   18 hours ago        Up About an hour                        ceos_vms6-1_VM0100
6846896eccb4        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0131
1957549bf246        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0130
7e1599906c99        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0129
e306a07e13e7        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0128
737f105579d3        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0127
3578dd337739        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0126
d8271c793088        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0125
a84de7d25aff        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0124
313d317908e3        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0123
97719d2a744d        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0122
e7cc3e14856a        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0121
36598386bf7d        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0120
12ac968175dd        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0119
fa86601464a0        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0118
089c8905b79a        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0117
c0e022e8cd92        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0116
547a0933c5dc        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0115
210ee065572b        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0114
ab198c2fa3ef        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0113
3e384332c741        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0112
d91010cea434        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0111
c23ce36d166a        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0110
a536f3999a7e        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0109
ce7a4f58da0d        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0108
c963807eeb64        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0107
c6288f60ee6e        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0106
f415db2a5a6d        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0105
85da75f0fcad        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0104
818f6d96520b        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0103
b5e1d9a97e27        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0102
bb122c8985bd        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0101
aadb1444e001        debian:jessie                                                "bash"                   31 hours ago        Up 31 hours                             net_vms6-1_VM0100
```

### CODE REVIEW

**Comment by lguohan** [comment]
obselete https://github.com/Azure/sonic-mgmt/pull/446/

File: ansible/templates/eos/t0-leaf.j2 [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ network {{ iface['ipv6'] }}
+{% endif %}
+{% endfor %}
+!
+management api http-commands
+ no protocol https
+ protocol http
+ no shutdown
+!
+end
```
Comment (wangxin) [inline ansible/templates/eos/t0-leaf.j2]: @lguohan 
The original ansible/roles/eos/templates/t0-leaf.j2 was renamed to ansible/templates/eos/t0-56-leaf.j2.
Content of this new ansible/templates/eos/t0-leaf.j2 is different with ansible/roles/eos/templates/t0-leaf.j2.
Static route configurations are missing in this new template.
Now the t0 topology does not have BGP routes. Is this intended or by mistake?

Comment (lguohan) [reply ansible/templates/eos/t0-leaf.j2]: it is intentional. I should have put more information int the pr to describe the motivation. Putting static route is difficult to maintain in long run, and also difficult to port over different vm (for example to use sonic vm as neighbor). the new design is the use arista vm as a passthrough, and the bgp route is generated using exabgp running inside the ptf docker. 
check this pr. https://github.com/Azure/sonic-mgmt/pull/1233

Comment (wangxin) [reply ansible/templates/eos/t0-leaf.j2]: Understood, thanks for the explanation.

**Pull request merged**